server/src/main/java/org/elasticsearch/index/mapper/BlockSourceReader.java

```diff
@@ -149,6 +175,41 @@ public String toString() {
         }
     }
 
+    private static class Geometries extends BlockSourceReader {
+
+        Geometries(ValueFetcher fetcher) {
+            super(fetcher);
+        }
+
+        @Override
+        protected void append(BlockLoader.Builder builder, Object v) {
+            if (v instanceof SpatialPoint point) {
+                BytesRef wkb = new BytesRef(WellKnownBinary.toWKB(new Point(point.getX(), point.getY()), ByteOrder.LITTLE_ENDIAN));
+                ((BlockLoader.BytesRefBuilder) builder).appendBytesRef(wkb);
+            } else if (v instanceof String wkt) {
+                try {
```

Comment:
We can generate WKB from the loaders so we don't need to do this, seems wasterful.

Reply:
Agreed, I'll look into changing this now.

Reply:
Done